Select Command

Yeehengg opened the pull request:
Context
-Added Select Parser 
- Edited Select Command

Links
#56

Yeehengg committed:
Work-In-Progess
- Not able to work
- Request to review codes

rickyyx commented on src/main/java/seedu/task/logic/commands/SelectCommand.java:
```diff
@@ -1,44 +1,74 @@
+			}
+		}
+			
+			EventsCenter.getInstance().post(new JumpToListRequestEvent(targetIndex - 1));
```
I guess the general logic is correct here. 
Do keep in mind that the select for task is actually implemented after we morphed the task card from person card. 

It is the event panel handling which u might need to add on, starting from posting a different but similar event. 

I suggest to take a look at task.ui.UIManager, as well as TaskListPanel and TaskListCard. 

Let's discuss about this feature tomorrow.

daredevil999 commented on src/main/java/seedu/task/logic/parser/SelectParser.java:
```diff
@@ -0,0 +1,111 @@
+
+	}
+	*/
+	private static final Pattern ITEM_INDEX_ARGS_FORMAT = Pattern.compile("(?<type>-t|-e)+(?<targetIndex>.+)");
```
Regex for index is something else.

daredevil999 commented on src/main/java/seedu/task/logic/parser/SelectParser.java:
```diff
@@ -0,0 +1,111 @@
+
+        Optional<Integer> index = parseIndex(args);
+        
+        if(!index.isPresent() && isTask){
```
isTask might not have been initialised. I think you have your order wrong. Should initialise first, before returning SelectCommand()

daredevil999 commented on src/main/java/seedu/task/logic/parser/SelectParser.java:
```diff
@@ -0,0 +1,111 @@
+     * Returns the specified index in the {@code command} IF a positive unsigned integer is given as the index.
+     *   Returns an {@code Optional.empty()} otherwise.
+     */
+    private Optional<Integer> parseIndex(String command) {
```
This way to obtain the index might be redundant, since index is always required for a valid Select command input. A more straightforward way would be to get index directly using the Pattern

daredevil999 reviewed (commented):
SelectParser requires some edits before it can be used in the ParserManager.

Yeehengg committed:
Added SelectParser
Added SelectTaskCommand
Added SelectEventCommand

Added JumpToEListRequestEvent
Rename JumpToListRequestEvent to JumpToTListRequestEvent

Refactor affected file with JumpToListRequestEvent to JumpToTListRequestEvent

Yeehengg committed:
Edited files for select command to work

rickyyx commented on src/main/java/seedu/task/logic/commands/SelectCommand.java:
```diff
@@ -1,44 +1,81 @@
+	public static final String MESSAGE_SELECT_EVENT_SUCCESS = "Selected Event: %1$s";
 
-        UnmodifiableObservableList<ReadOnlyTask> lastShownList = model.getFilteredTaskList();
+	sese
```
what's this line for?

Yeehengg replied:
my bad, typo typo. 
i will remove it and commit this file again

rickyyx commented on src/main/java/seedu/task/commons/events/ui/JumpToEListRequestEvent.java:
```diff
@@ -5,11 +5,11 @@
  * Indicates a request to jump to the list of items
  */
-public class JumpToListRequestEvent extends BaseEvent {
+public class JumpToEListRequestEvent extends BaseEvent {
```
i guess spelling out Event is better

Yeehengg replied:
will it be too long ? 
and repetitive for event to appear twice in the variable name

Yeehengg committed:
Refactor JumpToEListRequestEvent to JumpToEventListRequestEvent
Refactor JumpToTListRequestEvent to JumpToTaskListRequestEvent

Added affected files after refactoring

Yeehengg committed:
Removed typo error and commented codes

rickyyx committed:
resolve conflict
